tests/console/proxyscc_postreg.pm

```diff
@@ -0,0 +1,47 @@
+# Copyright (C) 2014-2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Do the registration against SCC after installation
+# Maintainer: Yi Xu <yxu@suse.de>
+
+use strict;
+use base "y2logsstep";
+
+use testapi;
+use registration;
+
+sub run {
+    return if get_var("HDD_SCC_REGISTERED");
+    my $version  = get_required_var("VERSION");
+    my $arch     = get_required_var("ARCH");
+    my $reg_code = get_required_var("SCC_REGCODE");
+    my $scc_url  = get_var("SCC_URL");
+
+    select_console 'root-console';
+    assert_script_run "SUSEConnect --url $scc_url -r $reg_code" if $scc_url;
+    assert_script_run "SUSEConnect -r $reg_code";
+    assert_script_run "SUSEConnect --list-extensions";
+
+    # add modules
+    assert_script_run "SUSEConnect -p sle-module-basesystem/$version/$arch";
+    assert_script_run "SUSEConnect -p sle-module-scripting/$version/$arch";
+    assert_script_run "SUSEConnect -p sle-module-desktop-applications/$version/$arch";
+    assert_script_run "SUSEConnect -p sle-module-development-tools/$version/$arch";
+    assert_script_run "SUSEConnect -p sle-module-server-applications/$version/$arch";
+    assert_script_run "SUSEConnect -p sle-module-legacy/$version/$arch";
```

Comment:
That won't fly. The list of modules must not be hardcoded here but depending on the product. Take a look into products/sle/main.pm for test variables where we already define the modules to be registered. Also I recommend to use a for-loop here to [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself).

Example:

```perl
for $module (qw(basesystem scripting…)) {
    assert_script_run "SUSEConnect -p sle-module-$module/$version/$arch";
}
```

and then replace the list with the content from the test variable accordingly.